src/main/java/org/broadinstitute/hellbender/engine/filters/ReadFilterLibrary.java

```diff
@@ -100,6 +114,7 @@ private ReadFilterLibrary(){ /*no instance*/ }
                     ! read.mateIsUnmapped() &&
                     read.mateIsReverseStrand() != read.isReverseStrand();}}
 
+    /** Filter out reads that does not align to the reference. */
```

Comment:
This filter is removing any read that does not have a CIGAR operator which consumes bases on the reference (that means, it does not align to the reference at all). For keep the read, it is enough if there is at least one of the following operators: M, D, N, =, X.

I do not know how to rephrase this for clarification...